src/Workspaces/CSharpTest/Formatting/FormattingTests.cs

```diff
@@ -10940,5 +10940,128 @@ public async Task TestNormalizeUsingAlias(string text, string expected)
         {
             await AssertFormatAsync(expected, text);
         }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/72196")]
+        public async Task TestSwitchExpression_WithFormattedListPattern_ShouldNotFormatListPattern()
```

Comment:
In theory, this test should not be necessary. Ideally each call to `AssertFormatAsync` should verify both of the following:

1. Formatting `code` produces `expected`
2. If `code` and `expected` are not the same, formatting `expected` also produces `expected`

Unfortunately, the `AssertFormatAsync` method is a bit convoluted to applying this change should be done in a separate pull request.